.github/workflows/basic.yml

```diff
@@ -12,13 +12,18 @@ jobs:
         build_type: [Debug, Release]
         compiler: [{c: gcc, cxx: g++}]
         pool_tracking: ['ON', 'OFF']
+        shared_library: ['OFF']
```

Comment:
Not for this PR: Now when we have this short version of flags variables, there is an information lost in GHA jobs listing. Example title of a job: 
[Build - Ubuntu (ubuntu-22.04, Release, gcc, g++, OFF, OFF)](https://github.com/oneapi-src/unified-memory-framework/actions/runs/7286710419/job/19855967077?pr=83#logs)
Who will know what does each of an `OFF` mean? We should print job matrix configuration somewhere during a build to distinguish failing jobs.

Reply:
We could always check the "configure build" section in logs where we see full cmake command. But still printing the config matrix is a good idea - please note that we have a task for this UMFW-173